libportability-gfx/src/impls.rs

```diff
@@ -1193,20 +1218,173 @@ pub extern "C" fn gfxDestroyFramebuffer(
     pRenderPass: *mut VkRenderPass,
 ) -> VkResult {
-    unimplemented!()
+    let info = unsafe { &*pCreateInfo };
+
+    // Attachment descriptions
+    let attachments = unsafe {
+        slice::from_raw_parts(info.pAttachments, info.attachmentCount as _)
+    };
+    let attachments = attachments
+        .into_iter()
+        .map(|attachment| {
+            assert_eq!(attachment.flags, 0); // TODO
+
+            let initial_layout = conv::map_image_layout(attachment.initialLayout);
+            let final_layout = conv::map_image_layout(attachment.finalLayout);
+
+            pass::Attachment {
+                format: conv::map_format(attachment.format),
+                ops: pass::AttachmentOps {
+                    load: conv::map_attachment_load_op(attachment.loadOp),
+                    store: conv::map_attachment_store_op(attachment.storeOp),
+                },
+                stencil_ops: pass::AttachmentOps {
+                    load: conv::map_attachment_load_op(attachment.stencilLoadOp),
+                    store: conv::map_attachment_store_op(attachment.stencilStoreOp),
+                },
+                layouts: initial_layout .. final_layout,
+            }
+        })
+        .collect::<Vec<_>>();
+
+    // Subpass descriptions
+    let subpasses = unsafe {
+        slice::from_raw_parts(info.pSubpasses, info.subpassCount as _)
+    };
+
+    // Store all attachment references, referenced by the subpasses.
+    let mut attachment_refs = Vec::with_capacity(subpasses.len());
+    struct AttachmentRefs {
+        input: Vec<pass::AttachmentRef>,
+        color: Vec<pass::AttachmentRef>,
+        resolve: Vec<pass::AttachmentRef>,
+        depth_stencil: Option<pass::AttachmentRef>,
+        preserve: Vec<usize>,
+    }
+
+    fn map_attachment_ref(att_ref: &VkAttachmentReference) -> pass::AttachmentRef {
+        (att_ref.attachment as _, conv::map_image_layout(att_ref.layout))
+    }
+
+    for subpass in subpasses {
+        let input = unsafe {
+            slice::from_raw_parts(subpass.pInputAttachments, subpass.inputAttachmentCount as _)
+                .into_iter()
+                .map(map_attachment_ref)
+                .collect()
+        };
+        let color = unsafe {
+            slice::from_raw_parts(subpass.pColorAttachments, subpass.colorAttachmentCount as _)
+                .into_iter()
+                .map(map_attachment_ref)
+                .collect()
+        };
+        let resolve = if subpass.pResolveAttachments.is_null() {
+            Vec::new()
+        } else {
+            unimplemented!()
+            /*
+            unsafe {
+                slice::from_raw_parts(subpass.pResolveAttachments, subpass.colorAttachmentCount as _)
+                    .into_iter()
+                    .map(map_attachment_ref)
+                    .collect()
+            }
+            */
+        };
+        let depth_stencil = if subpass.pDepthStencilAttachment.is_null() {
```

Comment:
let's use [as_ref](https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.as_ref) with `map` for clarity